csharp/src/Apache.Arrow.Adbc/AdbcDriver11.cs

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace Apache.Arrow.Adbc
+{
+    /// <summary>
+    /// This provides a common interface for vendor-specific driver
+    /// initialization routines.
+    /// </summary>
+    public abstract class AdbcDriver11 : IDisposable
```

Comment:
This class _should_ inherit from `AdbcDriver`. Though, if we ever have a Driver Manager, it _will_ have to inherit from `AdbcDriver`.

Reply:
Sorry for not being clearer; the intent is to entirely replace the existing classes with these "11" versions. But I don't want to do it in one gigantic pull request; I want to stage the changes to make them digestable.

Reply:
Sorry for the misunderstanding. I'll withdraw the inheritance related comments